sap/5.1/modules/ROOT/pages/sap-connector-troubleshooting.adoc

```diff
@@ -0,0 +1,364 @@
+
+[[common-errors]]
+== Common Errors
+
+[[error-jco-missing-libs]]
+=== Missing JCo Libraries Errors
+
+These errors typically occur when you test the connection or try to deploy the Mule application without loading one or more of the required SAP JCo dependencies:
+
+* Missing JCo native library
++
+[source,text,linenums]
+----
+java.lang.ExceptionInInitializerError: JCo initialization failed with java.lang.UnsatisfiedLinkError: no sapjco3 in java.library.path
+----
++
+[.center.text-center]
+image::sap-error-jco-libs1.png[SAP Error JCo Native Library]
+
+* Missing JCo JAR
++
+----
+java.lang.NoClassDefFoundError: com/sap/conn/jco/JCoTraceListener
+----
++
+[.center.text-center]
+image::sap-error-jco-libs2.png[SAP Error JCo JAR]
+
+* Missing IDoc JAR
++
+----
+java.lang.NoClassDefFoundError: com/sap/conn/idoc/IDocMetaDataUnavailableException
+----
++
+[.center.text-center]
+image::sap-error-jco-libs3.png[SAP Error IDoc JAR]
+
+To resolve this error:
+
+. Click the *Add File* button next to the dependency with the red exclamation mark icon in the *Required dependencies* section.
+. Browse through the dependencies and select the appropriate file. +
+The missing dependencies are automatically added to the project's classpath.
+
+[TIP]
+In *Package Explorer*, right-click the project and navigate to *Build Path > Configure Build Path* to see the libraries in the project's build path.
+
+[[error-jco-classloader-conflicts]]
+=== JCo Classloader Conflicts
+
+When you load the native library from the global configuration, a copy of the file is placed inside `$YOUR_APP/src/main/app/lib`, but the source file is not removed; hence, you get the following exception if it shares the same directory as the JCo JAR files when testing the connection or deploying your app:
+
+[source,text,linenums]
+----
+java.lang.ExceptionInInitializerError: JCo initialization failed with java.lang.UnsatisfiedLinkError: Native Library /home/mule/sap-errors/lib/jco/libsapjco3.so already loaded in another classloader
+----
+
+[.center.text-center]
+image::sap-error-jco-classloader-folder.png[SAP Error JCo Classloader Folder]
+
+To resolve this error:
+
+Choose either of the following solutions. The first is the simplest, but the second is the best practice approach.
+
+* Remove the _duplicate_ native library file from the directory where your JCo JAR files reside:
+
+[.center.text-center]
+image::sap-error-jco-classloader-fix.png[SAP Error JCo Classloader Fix]
+
+* Configure the environment variable `LD_LIBRARY_PATH` to hold the dynamic link library and share it across multiple applications deployed within the same Mule runtime server.
+
+
+[[error-jco-version-conflicts]]
+=== JCo Version Conflicts
+
+The most frequent cause of the following error is that different versions of the native library and JCo JAR files are being employed:
```

Comment:
IMHO, "used" is a better word than "employed," especially considering localization